pkg/kubectl/pdb.go

```diff
@@ -95,7 +102,119 @@ func (s *PodDisruptionBudgetV1Generator) validate() error {
 		return fmt.Errorf("a selector must be specified")
 	}
 	if len(s.MinAvailable) == 0 {
-		return fmt.Errorf("the minimim number of available pods required must be specified")
+		return fmt.Errorf("the minimum number of available pods required must be specified")
+	}
+	return nil
+}
+
+// PodDisruptionBudgetV2Generator supports stable generation of a pod disruption budget.
+type PodDisruptionBudgetV2Generator struct {
+	Name           string
+	MinAvailable   string
+	MaxUnavailable string
+	Selector       string
+}
+
+// Ensure it supports the generator pattern that uses parameters specified during construction.
+var _ StructuredGenerator = &PodDisruptionBudgetV2Generator{}
+
+func (PodDisruptionBudgetV2Generator) ParamNames() []GeneratorParam {
```

Comment:
Why do we need to implement the [`Generator interface`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate.go#L39)? @kargakis 
I think we are using `StructuredGenerator interface` now.